src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatReader.cs

```diff
@@ -138,7 +138,32 @@ private void ReflectionSetMemberValue(object obj, object memberValue, MemberInfo
                 else
                 {
                     object property = propInfo.GetValue(obj);
-                    MethodInfo addMethod = property.GetType().GetMethod("Add");
+                    Type propertyType = property.GetType();
+                    MethodInfo addMethod = propertyType.GetMethod("Add");
```

Comment:
The fix was not to address the issue with the occurrence of multiple Add() methods. The issue was that the above `propertyType.GetMethod("Add")` cannot get the Add() method that a type inherits explicitly from `ICollection<T>`. So the following code was to get the Add the method from ICollection.

I just debug the code, however, I found the entire else block was not hit at all. The code was re-routed to other places by other fixes. I am going to delete this else block.